### PR TITLE
Yakushev 24 06 filter feature

### PR DESCRIPTION
**Pull request by PRO100Spike**
Добавил фильтры по дате и тексту + поиск по дате и тексту

### CODE REVIEW

**Review by FominSergey5869 — APPROVED** [review]

File: src/components/posts/element.jsx [hunk 1]
```diff
@@ -65,19 +82,28 @@ const GroupElement = memo(
...
-                    }
-                    onClick={() => addFilter(filter.type)}
-                  ></Button>
+                  <>
+                    <Button
+                      key={filter.type}
+                      value={filter.name}
+                      type={
+                        filter.condition
+                          ? filter.condition === 1
```
Comment (khigor777) [inline src/components/posts/element.jsx:119]: All digits should be moved to constants

File: src/utils/filter-feed.js [hunk 2]
```diff
@@ -7,37 +7,53 @@ export const filterTypeEnum = {
...
+      case 'video':
+        if (filter.condition > 0)
+          filteredFeed = filteredFeed.filter((post) => {
+            if (!post.attachments) return false
 
-          if (filter.condition === 1) {
-            for (let i = 0; i < post.attachments.length; i++) {
-              if (post.attachments[i].type === filter.type) {
-                return true
+            if (filter.condition === 1) {
```
Comment (khigor777) [inline src/utils/filter-feed.js:20]: All digits should be moved to constants

File: src/utils/filter-feed.js [hunk 3]
```diff
@@ -7,37 +7,53 @@ export const filterTypeEnum = {
...
+              return false
             }
-            return false
-          }
 
-          if (filter.condition === 2) {
-            for (let i = 0; i < post.attachments.length; i++) {
-              if (post.attachments[i].type === filter.type) {
-                return false
+            if (filter.condition === 2) {
```
Comment (khigor777) [inline src/utils/filter-feed.js:29]: All digits should be moved to constants

File: src/utils/filter-feed.js [hunk 4]
```diff
@@ -7,37 +7,53 @@ export const filterTypeEnum = {
...
-          }
 
-          return false
+            return false
+          })
+        break
+      case 'text-search':
+        break
+      case 'text':
+        if (filter.condition > 0) {
```
Comment (khigor777) [inline src/utils/filter-feed.js:58]: All digits should be moved to constants

File: src/utils/filter-feed.js [hunk 5]
```diff
@@ -7,37 +7,53 @@ export const filterTypeEnum = {
...
-      })
-    } else {
-      if (filter.condition > 0)
-        filteredFeed = filteredFeed.filter((post) => {
-          if (!post.attachments) return false
+    switch (filter.type) {
+      case 'photo':
+      case 'link':
+      case 'video':
+        if (filter.condition > 0)
```
Comment (khigor777) [inline src/utils/filter-feed.js:16]: All digits should be moved to constants

File: src/redux/group-reducer.js [hunk 6]
```diff
@@ -55,15 +57,16 @@ const groupReducer = (state = initialState, action) => {
...
       return stateCopy
     case ADD_FILTER:
+      console.log(action.payload)
       const filterList = [...state.filterList]
       let condition = filterList.find(
-        (e) => e.type === action.payload
+        (e) => e.type === action.payload.type
       ).condition
 
       if (condition === 2) condition = 0
```
Comment (khigor777) [inline src/redux/group-reducer.js:68]: All digits should be moved to constants

**Review by khigor777 — CHANGES_REQUESTED** [review]

**Review by khigor777 — APPROVED** [review]

**Pull request merged**